shipctl/x86_64/WindowsServer_2016/utility.ps1

```diff
@@ -728,3 +786,106 @@ function _send_notification() {
     }
   }
 }
+
+function _send_irc_notification() {
+  param(
+    [string]$server,
+    [string]$channel,
+    [string]$nick,
+    [string]$password,
+    [string]$payload
+  )
+  $port = 6667
+
+  # make the connection
+  $client = New-Object -TypeName Net.Sockets.TcpClient
+  $client.Connect($server, $port)
+  $netStream = $client.GetStream()
+  $encoding = [System.Text.Encoding]::ASCII
+  $streamWriter = New-Object -Type System.IO.StreamWriter -ArgumentList $netStream, $encoding
+  $connection = "on"
+  # send initial login commands
+  $streamWriter.WriteLine("PASS $password")
+  $streamWriter.Flush()
+  $streamWriter.WriteLine("NICK $nick")
+  $streamWriter.WriteLine("USER $env:UserName 0 * :Shippable Assembly Lines")
+  $streamWriter.Flush()
+
+  # read data
+  $maxWaitTime = 60 # seconds
+  $message = ""
+  $joined = $false
+  while (($connection -eq "on") -and ($maxWaitTime -ge 0)) {
+
+    if ($netStream.DataAvailable) {
+      [char]$char = $netStream.ReadByte()
+      if ($char -eq 13) {
+        write-output "$message"
+        $joined = _check_message $streamWriter $message $channel
+        if ($joined) {
+        }
```

Comment:
I'm guessing something should happen here...

Reply:
i'll delete this. the condition is handled separately later. i think i had a debug print statement there that i removed but left the empty condition :)